sql/_13_issues/_24_2h/cases/cbrd_25316_1.sql

```diff
@@ -0,0 +1,159 @@
+select '' as "test case1: use count";
+update /*+ recompile */ tbl_b b set b.c_r = count(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use sum";
+update /*+ recompile */ tbl_b b set b.c_r = sum(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use min";
+update /*+ recompile */ tbl_b b set b.c_r = min(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use max";
+update /*+ recompile */ tbl_b b set b.c_r = max(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use stddev";
+update /*+ recompile */ tbl_b b set b.c_r = stddev(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use stddev_pop";
+update /*+ recompile */ tbl_b b set b.c_r = stddev_pop(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use stddev_samp";
+update /*+ recompile */ tbl_b b set b.c_r = stddev_samp(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use var_pop";
+update /*+ recompile */ tbl_b b set b.c_r = var_pop(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use var_samp";
+update /*+ recompile */ tbl_b b set b.c_r = var_samp(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use variance";
+update /*+ recompile */ tbl_b b set b.c_r = variance(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use ntile";
+update /*+ recompile */ tbl_b b set b.c_r = ntile(4) over (partition by b.col_a order by b.col_b);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use median";
+update /*+ recompile */ tbl_b b set b.c_r = median(b.col_b) over (partition by b.col_a);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use first_value";
+update /*+ recompile */ tbl_b b set b.c_r = first_value(b.col_b) over (partition by b.col_a order by b.col_b);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use last_value";
+update /*+ recompile */ tbl_b b set b.c_r = last_value(b.col_b) over (partition by b.col_a order by b.col_b);
+select distinct col_a, to_char(c_r) from tbl_b order by col_a;
+select '' as "test case1: use nth_value";
+update /*+ recompile */ tbl_b b set b.c_r = nth_value(b.col_b, 2) over (partition by b.col_a order by b.col_b);
```

Comment:
test case1에서는 nth_value(b.col_b, 2)를 조회하고 있는데
test case3, 4에서는 nth_value(b.col_b, 1)을 조회하고 있습니다.
동일한 nth_value 값을 업데이트 하기 위해서 아래와 같이 변경하면 좋겠습니다.
```suggestion
update /*+ recompile */ tbl_b b set b.c_r = nth_value(b.col_b, 1) over (partition by b.col_a order by b.col_b desc);
```

Reply:
리뷰 의견을 반영해서 [CBRD-25316 ](http://jira.cubrid.org/browse/CBRD-25316)이슈에 cbrd-25316_test-case_20240923.zip를 첨부했습니다.
검토 부탁드립니다.